common/fixation_entry.go

```diff
@@ -320,18 +358,12 @@ func (fs *FixationStore) PutEntry(ctx sdk.Context, index string, block uint64) {
 		panic("PutEntry with unknown index: " + index)
 	}
 
-	if entry.Refcount == 0 {
-		panic("PutEntry with refcount zero index: " + index)
-	}
-
-	entry.Refcount -= 1
-
-	if entry.Refcount == 0 {
-		// never overflows because because ctx.BlockHeight is int64
-		entry.StaleAt = uint64(ctx.BlockHeight()) + uint64(types.STALE_ENTRY_TIME) - 1
+	if entry.Block != block {
+		panic("PutEntry with block mismatch index: " + index +
+			" got " + strconv.Itoa(int(entry.Block)) + " expected " + strconv.Itoa(int(block)))
 	}
 
-	fs.setEntry(ctx, entry)
+	fs.putEntry(ctx, entry)
```

Comment:
why do we need an additional putEntry? why not put all the code in one function?